### PR TITLE
Unify the unit for different processors

### PR DESCRIPTION
**Pull request by BoxiLi**
Unify the unit of parameters. All the parameters should be given in the unit of frequency (GHz, MHz etc), and will be saved in the same unit (instead of been multiplied by 2pi as before). The operators are multiplied by 2 pi to produce the correct result.

### CODE REVIEW

File: src/qutip_qip/device/spinchain.py [hunk 1]
```diff
@@ -52,44 +52,53 @@ class SpinChain(ModelProcessor):
...
-        each qubit. A list of size `num_qubits` or a float for all qubits.
+        each qubit. A list of size ``num_qubits`` or a float for all qubits.
+
+    **params:
+        Keyword argument for hardware parameters, in the unit of frequency
+        (MHz, GHz etc, the unit of time list needs to be adjusted accordingly).
+        Qubit parameters can either be a float or an array of the length
+        ``num_qubits``.
+        ``sxsy``, should be either a float or an array of the length
+        ``num_qubits-1`` (for :class:`.LinearSpinChain`) or ``num_qubits``
```
Comment (nathanshammah) [inline src/qutip_qip/device/spinchain.py:82]:
```suggestion
        ``num_qubits-1`` (for :class:`.LinearSpinChain`) or ``num_qubits``
```
```suggestion
        ``num_qubits=1`` (for :class:`.LinearSpinChain`) or ``num_qubits``
```

File: src/qutip_qip/device/spinchain.py [hunk 2]
```diff
@@ -283,62 +264,33 @@ def topology_map(self, qc):
...
-        The interaction strength for each of the qubit pair in the system.
-
-    t1: list or float, optional
-        Characterize the decoherence of amplitude damping for
-        each qubit.
-
-    t2: list of float, optional
-        Characterize the decoherence of dephasing for
-        each qubit.
+    Spin chain model with circular topology. See :class:`.SpinChain`
```
Comment (nathanshammah) [inline src/qutip_qip/device/spinchain.py:267]: Include the `Parameters` blurb though?

File: src/qutip_qip/device/cavityqed.py [hunk 3]
```diff
@@ -75,32 +75,31 @@ class DispersiveCavityQED(ModelProcessor):
...
 
+    **params:
+        Keyword argument for hardware parameters, in the unit of GHz.
+        Qubit parameters can either be a float or a list of the length
+        ``num_qubits``.
+
+        - ``deltamax``: the pulse strength of sigma-x control, default ``1.0``
+        - ``epsmax``: the pulse strength of sigma-z control, default ``9.5``
+        - ``eps``: the epsilon for each of qubit, default ``10``
+        - ``delta``: the delta for each of qubit, default ``0.0``
```
Comment (nathanshammah) [inline src/qutip_qip/device/cavityqed.py:94]:
```suggestion
        - ``eps``: the epsilon for each of qubit, default ``10``
        - ``delta``: the delta for each of qubit, default ``0.0``
```
Can you add something more descriptive here for `eps` and `delta`?

**Review by nathanshammah — CHANGES_REQUESTED** [review]
Added a couple of comments. Can you add some information about the `\pi` convention somewhere?

File: src/qutip_qip/device/spinchain.py (hunk 1 above)
Comment (BoxiLi) [reply src/qutip_qip/device/spinchain.py:82]: No, it is `num_qubits-1`, for LinearSpinChain the list of coupling strength is one shorter than the number of qubits

**Pull request merged**